checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -81,6 +131,35 @@ public void handle(MethodInvocationTree tree, AnnotatedExecutableType method) {
         }
     }
 
+    /**
+     * Determine whether the argument is a field access expression of which the declaration has a
```

Comment:
This is a description of the implementation strategy, but it is not a specification of the method.

I think the specification is:
 * return true if the given expression is an empty array (has type `@ArrayLen(0)`), using heuristics

Is that correct?
The current implementation only works for field accesses, but that is not worth putting in the Javadoc nor embedding in the name.  These changes will make the code clearer.

Also, the addition of this method makes the documentation and name of `isHandledArrayCreation` inaccurate.  It no longer handles all the heuristics.  (It does handle just some of the 0-length arrays.)  Why isn't the code `trustArrayLenZero && isArrayLenZeroFieldAccess(argument)` in the body of `isHandledArrayCreation`?  If you don't want to do that, then it seems that restructuring and renaming it is necessary.